tests/qa_automation/acceptance.pm

```diff
@@ -64,9 +68,29 @@ sub run() {
 
     # QA DB upload happens for each module
     # output the failed tests to serial console
-    type_string "export PS1=#\n";
+    assert_script_run "export PS1=#";
+    # qa_testset_automation remove the oldlogs, we create it back here for junit collecting information
+    assert_script_run "mkdir /var/log/qa/oldlogs; cd /var/log/qa/oldlogs; for i in /var/log/qaset/log/*.bz2; do tar -xjvf \$i; done";
     type_string "find /var/log/qa/oldlogs/ -name test_results | xargs grep -l -B1 ^1 | tee -a /dev/$serialdev\n";
     assert_screen 'no_output_from_qa_find';
+
+    # test junit
+    my $testset = get_var('QA_TESTSET');
+    if ($testset == "acceptance") { 
+        my $type = "stress_validation";
+    }
+    elsif  ($testset == "regression") {
+        my $type = "user_regression";
+    }
+    elsif  ($testset == "kernel") {
+        my $type = "kernel_regression";
+    }
+    else {
+        my $type = "all";
+    }
+    assert_script_run "/usr/share/qa/qaset/bin/junitxml_generator.py -t " . $type . " -l /var/log/qaset/runs/ -s /var/log/qaset/submission/ -o /tmp/junit.xml";
```

Comment:
there is no need to concat variables like this. You can just use "-t $type" - perl expands variables in strings